### PR TITLE
add support Orbbec Gemini2 and Gemini2 XL camera

### PR DESCRIPTION
**Pull request by zzuliys**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch

### CODE REVIEW

File: samples/cpp/videocapture_obsensor.cpp [hunk 1]
```diff
@@ -21,6 +21,11 @@ int main()
     Mat image;
     Mat depthMap;
     Mat adjDepthMap;
+
+    // 深度最小值
+    const double minVal = 300;
+    // 深度最大值
+    const double maxVal = 5000;
```
Comment (WanliZhong) [inline samples/cpp/videocapture_obsensor.cpp:28]: Chinese comments should be removed

**Comment by zzuliys** [comment]
> ```
> samples/cpp/videocapture_obsensor.cpp:41: trailing whitespace.
> +            
> samples/cpp/videocapture_obsensor.cpp:43: trailing whitespace.
> +            {           
> ```

Extra spaces have been removed

File: samples/cpp/videocapture_obsensor.cpp (hunk 1 above)
Comment (zzuliys) [reply samples/cpp/videocapture_obsensor.cpp]: The Chinese comment has been removed

**Comment by zzuliys** [comment]
@WanliZhong Add note that some Orbbec cameras do not support Linux kernels version.Please check.

**Review by WanliZhong — APPROVED** [review]
As the comment says: "Astra2, Gemini2, and Gemini2L cameras currently only support Windows and Linux kernel versions 4.15, and higher versions of Linux kernel may have exceptions"

The issue will be fixed after 4.9.0 in a separate PR

File: CMakeLists.txt [hunk 2]
```diff
@@ -468,6 +467,7 @@ OCV_OPTION(WITH_ONNX "Include Microsoft ONNX Runtime support" OFF
 OCV_OPTION(WITH_TIMVX "Include Tim-VX support" OFF
   VISIBLE_IF TRUE
   VERIFY HAVE_TIMVX)
+# attention: Astra2, Gemini2, and Gemini2L cameras currently only support Windows and Linux kernel versions <= 4.15, and higher versions of Linux kernel may have exceptions.
```
Comment (asmorkalov) [inline CMakeLists.txt:470]: The statement about kernel version is not clear. Do you mean, that camera requires kernel 4.15 or older?
Please add reference to the camera documentation, ticket or whatever to track support status.

File: modules/videoio/include/opencv2/videoio.hpp [hunk 3]
```diff
@@ -128,7 +128,7 @@ enum VideoCaptureAPIs {
        CAP_INTEL_MFX    = 2300,         //!< Intel MediaSDK
        CAP_XINE         = 2400,         //!< XINE engine (Linux)
        CAP_UEYE         = 2500,         //!< uEye Camera API
-       CAP_OBSENSOR     = 2600,         //!< For Orbbec 3D-Sensor device/module (Astra+, Femto)
+       CAP_OBSENSOR     = 2600,         //!< For Orbbec 3D-Sensor device/module (Astra+, Femto, Astra2, Gemini2, Gemini2L, Gemini2XL, Femto Mega) attention: Astra2, Gemini2, and Gemini2L cameras currently only support Windows and Linux kernel versions 4.15, and higher versions of Linux kernel may have exceptions
```
Comment (asmorkalov) [inline modules/videoio/include/opencv2/videoio.hpp:131]: The same.

Comment (zzuliys) [reply modules/videoio/include/opencv2/videoio.hpp]: I have resolved this issue.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**